classes/event/observer.php

```diff
@@ -38,7 +38,8 @@ public static function course_updated(\core\event\course_updated $event) {
 
         $courseid = $event->objectid;
         $crawler = new \tool_crawler\robot\crawler();
-        return $crawler->mark_for_crawl($CFG->wwwroot . '/', 'course/view.php?id=' . $courseid, $courseid);
+        return $crawler->mark_for_crawl($CFG->wwwroot . '/', 'course/view.php?id=' . $courseid, $courseid,
```

Comment:
manually crafting these urls can be problemattic with certain edge cases (like mod_label). I think it would be better to defer to something else which builds them consistently, eg context_helper::get_url()

You might find you can refactor all of the handlers into the same handle then

Reply:
I've changed these to use the context_course and context_module helpers, and use get_url->out